src/core/probe/builder.rs

```diff
@@ -37,6 +49,72 @@ pub(super) fn reuse_map_fds(
     Ok(())
 }
 
+/// Replaces raw bpf programs using both btf-rs and bpf_sys
+/// facilities. This is currently needed because libbpf-rs doesn't
+/// have loading nor replacing capabilities for non-ELF files.
+fn replace_raw_filter(filter: &FilterPacket, fd: u32, tgt_func: &str) -> Result<()> {
```

Comment:
uhm, makes sense, in general, to keep it more generic (before this respin it was, and yeah, and the current is probably an oversimplification before a rework), although I don't think the raw function will be consumed by other filters. Let's step back and avoid tying it to a single filter type.

About your proposal, I wonder if something like this could be enough:

```
type BpfProg = Vec<u8>;

struct BpfFilter {
    prog: BpfProg,
    target: String,
}

enum Filter {
    Packet(BpfFilter),
    Sock(BpfFilter),
    Dev(BpfFilter),
}
```

I guess it should be enough to represent replaceable filters with `BpfFilter`.
If you see any particular value in struct FilterPacket(BpfFilter);

Reply:
I'm respinning only for convenience with the above (so it's easier to look at it), but please let's continue this discussion. It's fairly easy to change it.